Zotero/Scenes/Detail/PDF/ViewModels/PDFReaderActionHandler.swift

```diff
@@ -2135,8 +2136,17 @@ final class PDFReaderActionHandler: ViewModelActionHandler, BackgroundDbProcessi
 
         DDLogInfo("PDFReaderActionHandler: database annotation changed")
 
-        // Get sorted database keys
-        var keys = (viewModel.state.snapshotKeys ?? viewModel.state.sortedKeys).filter({ $0.type == .database })
+        // Keys for deletions and modifications are indexed by the previously observed items
+        var previousObservedKeys: [PDFReaderState.AnnotationKey] = []
+        for item in viewModel.state.databaseAnnotations {
+            previousObservedKeys.append(.init(key: item.key, type: .database))
+        }
+        // Keys for insertions are indexed by the currently observed items
+        var currentObservedKeys: [PDFReaderState.AnnotationKey] = []
```

Comment:
Does this even need to be created? I only see `currentObservedKeys.count` being used, so you could just do `objects.count` or store `let currentKeysCount = objects.count` and use that for clarity.

Reply:
Check https://github.com/zotero/zotero-ios/pull/999/commits/6e0b28fbb1a3b464b94c6cc4bbe698c5c366c8cb